### DISCUSSION TITLE
A place to download and upload language files ...

### DISCUSSION THREAD

**Discussion by rudydepo** [geodesicsolutions-community/geocore-community#213] [Ideas]
A place to download and upload language files ...
It's a massive work to translate manually (Google Translate is not that good in some languages)
Dutch almost avilable if needed.
French will available in 2 weeks.
German in a month.

**Comment by blufyremedia** [comment]
Hello @rudydepo,

The platform already has language capabilities built-in, with the ability to export and import dumps of all site verbiage. This covers system verbiage, but not user-specified content, such as listing titles or descriptions.

You are correct in that Google translate is not perfect, and not an ideal solution - however it does provide a more cost effective means of offering alternate languages. Exporting your site verbiage, translating, and then importing and configuring the translated verbiage would also enable you to offer the language directly on your site, rather than relying on the user to translate using a widget or browser extension.

Alternate language files used to be provided with the platform for a handful of languages, however this was _ineffective_, as most sites have their **own nuances and adjusted verbiage**, rendering some, or most, of a "pre-made" language file obsolete/inaccurate.

Beyond this, was there anything specific you were looking for / inquiring about?

**Comment by 1kiwi1** [comment]
> A place to download and upload language files ... It's a massive work to translate manually (Google Translate is not that good in some languages) Dutch almost avilable if needed. French will available in 2 weeks. German in a month.

Hi again rudydepo.  Once upon a time Geodesicsolutions offered language packs to run a single or multi-lingual site.  There were 20+ language packs for client side use, you can see the archive here: **https://web.archive.org/web/20170828011410/http://geodesicsolutions.com/add-ons/free-add-ons/386-language-packs.html**.  People actually need site language translation templates, they are a great starting point to modify for personal use.  Generous users donated their work to save others reinventing the wheel, so you could edit them to your own preference.  If you are building language packs there will be members who will benefit from the hard work, thank you!  If you need assistance, I have most of the language packs from GeodesicSolutions in zip.  For your circumstances there is available default/fusion/marquee in Dutch, French and German.   You could harness them for what your doing.  Just message me if you require them.  I'll make them available sometime at our site.  Thanks!

**Reply by vicos59** [reply]
If you would like to submit the files, I'm sure we can include them in the distro. We didn't have them, which is why they're not there now. 

Jon indicated that they were never originally included in the company's git files: https://github.com/geodesicsolutions-community/geocore-community/issues/86#issuecomment-1049878796

**Reply by blufyremedia** [reply]
Correct, the old translation packs are quite outdated and obsolete, so they aren't very relevant to the current build and were not a priority. If someone would like to work on those and bring them up to date as generic translations, they can certainly be included here. 

While they could be a suitable starting point, these were based on general "out of the box" verbiage, which again, is not typically relevant for most sites, unless one hasn't done much to customize.  Exporting your site verbiage (after your customizations), will give you a more accurate .CSV that you can work off of for relevant translations to your specific needs.

@1kiwi1 If you would like to contribute them as legacy resources, please do so - this community is for supporting users and providing the resources needed in one place.

**Reply by 1kiwi1** [reply]
> Correct, the old translation packs are quite outdated and obsolete, so they aren't very relevant to the current build and were not a priority. If someone would like to work on those and bring them up to date as generic translations, they can certainly be included here.
> 
> While they could be a suitable starting point, these were based on general "out of the box" verbiage, which again, is not typically relevant for most sites, unless one hasn't done much to customize. Exporting your site verbiage (after your customizations), will give you a more accurate .CSV that you can work off of for relevant translations to your specific needs.
> 
> @1kiwi1 If you would like to contribute them as legacy resources, please do so - this community is for supporting users and providing the resources needed in one place.

@blufyremedia ... Although this edit/comment is slightly delayed, it remains relevant.  I wanted to provide an update on the previous discussion regarding custom language builds.  Geocore software indeed supports uploading CSV files via its admin interface for the purpose of multi language editing.  Please note that this GitHub build of Community Geodesic Software in use is based on the legacy core version 18 from Geodesic Solutions. There have been no template updates applied beyond version 18, including for version 20.  Consequently, the four default template sets provided are from Geodesic’s original v18 build, created around 2016, and use the class="no-js design2016" design.  This aligns with the versioning timeline of the current project’s templates.  It is also important to highlight that the language template sets or csv.txt files last updated by Geodesic are not outdated; they correspond to the default template sets with the class="no-js design2016" design.  These are useful if you intend to use a default theme along with the provided translation, especially for standard, generic code.

Best regards,
